### PR TITLE
fix: [modules] Gracefully accept no hashlookup fixes #916

### LINKED ISSUE #916: [Bug]: Lookyloo throws LookylooException when hashlookup is not enabled in config
### What happened?

When the modules config defines `enabled": false,` in the Hashlookup section, and the user asks the webUI to show the hashlookups, the webui does not show an error message, and the code throws a LookylooException.

### If the issue is related to the web interface, what browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by cvandeplas**
## Type of change

**Description:**
Fixes bug #916 - accept hashlookup being disabled

**Select the type of change(s) made in this pull request:**
- [x] Bug fix *(non-breaking change which fixes an issue)*
- [ ] New feature *(non-breaking change which adds functionality)*
- [ ] Documentation *(change or fix to documentation)*

---------------------------------------------------------------------------------------------------------

Fixes #916

## Proposed changes 

* cleanly show the message to the user instead of throwing an error in the code.

### CODE REVIEW

**Comment by cvandeplas** [comment]
Note: the PR also contains minor changes in the redis script.
Note2:  commenting out the following lines in `indexing.conf` and `cache.conf` is needed for redis 7.0.x to start properly.
- `# locale-collate ""`
- `# set-max-listpack-entries 128`
- `# set-max-listpack-value 64`

File: cache/run_redis.sh [hunk 1]
```diff
@@ -5,6 +5,8 @@ set -x
 
 if [ -f  ../../valkey/src/valkey-server ]; then
     ../../valkey/src/valkey-server ./cache.conf
-else
+elif [ -f ../../redis/src/redis-server ]; then
     ../../redis/src/redis-server ./cache.conf
+else
+    /usr/bin/redis-server ./cache.conf
```
Comment (Rafiot) [inline cache/run_redis.sh:11]: I'm not totally against that, but I always use the latest version of redis (or valkey now), and that includes the latest commands. Using the version from the packages often gives you a very old version that may or may not work.

Can you print a message saying that it is unsupported and may cause runtime issues?

File: indexing/run_redis.sh [hunk 2]
```diff
@@ -5,6 +5,9 @@ set -x
 
 if [ -f  ../../valkey/src/valkey-server ]; then
     ../../valkey/src/valkey-server ./indexing.conf
-else
+elif [ -f ../../redis/src/redis-server ]; then
     ../../redis/src/redis-server ./indexing.conf
+else
+    /usr/bin/redis-server ./indexing.conf
```
Comment (Rafiot) [inline indexing/run_redis.sh:11]: I'm not totally against that, but I always use the latest version of redis (or valkey now), and that includes the latest commands. Using the version from the packages often gives you a very old version that may or may not work.

Can you print a message saying that it is unsupported and may cause runtime issues?

Comment (cvandeplas) [reply indexing/run_redis.sh:12]: I added the message for both cases. Also (tried) to fix the python test raising the variable type. Can you approve test workflows again?

File: website/web/__init__.py [hunk 3]
```diff
@@ -962,10 +962,14 @@ def urls_rendered_page(tree_uuid: str) -> WerkzeugResponse | str | Response:
...
-    # We only want unique URLs for the template
-    for sha1, entries in merged.items():
-        entries['nodes'] = {node.name for node in entries['nodes']}
+    try:
+        merged, total_ressources = lookyloo.merge_hashlookup_tree(tree_uuid)
+        # We only want unique URLs for the template
+        for sha1, entries in merged.items():
+            entries['nodes'] = {node.name for node in entries['nodes']}
+    except Exception:  # error or module not enabled
+        merged = []
```
Comment (Rafiot) [inline website/web/__init__.py:971]: Now that I'm thinking about it, no need to render the `hashlookup.html` template there, you can simply render the tree flash an error message like that:

https://github.com/Lookyloo/lookyloo/blob/b6ff78f84985a078d2750716397bf119494dc1eb/website/web/__init__.py#L958-L962

**Pull request merged**